external-crates/move/crates/move-core-types/src/annotated_extractor.rs

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    account_address::AccountAddress, annotated_value as A, annotated_visitor as AV,
+    language_storage::TypeTag,
+};
+
+#[derive(Debug, Clone)]
+pub enum Element<'e> {
+    Field(&'e str),
+    Index(u64),
+    Type(&'e TypeTag),
```

Comment:
Chatted about this synchronously. If you could just add a comment separating out "selectors" from "filters"